homeassistant/components/vegehub/config_flow.py

```diff
@@ -0,0 +1,192 @@
+            errors["base"] = "cannot_connect"
+
+        if self._hub is None:
+            # Show the form to allow the user to manually enter the IP address
+            return self.async_show_form(
+                step_id="user",
+                data_schema=vol.Schema(
+                    {
+                        vol.Required(CONF_IP_ADDRESS): str,
+                    }
+                ),
+                errors={},
+            )
+
+        # If we already have an IP address, we can just ask the user if they want to continue
+        return self.async_show_form(step_id="user", errors=errors)
+
+    async def async_step_zeroconf(
+        self, discovery_info: zeroconf.ZeroconfServiceInfo
+    ) -> ConfigFlowResult:
+        """Handle zeroconf discovery."""
+
+        # Extract the IP address from the zeroconf discovery info
+        device_ip = discovery_info.host
+
+        # Keep track of which IP addresses have already had their MAC addresses
+        # discovered. This allows us to skip the MAC address retrieval for devices
+        # that don't need it. This stops us from waking up a hub every time we see
+        # it come on-line.
+        have_mac = False
+        if device_ip in ip_dict:
+            have_mac = True
+
+        self._hostname = discovery_info.hostname.removesuffix(".local.")
+        self._config_url = (
+            f"http://{discovery_info.hostname[:-1]}:{discovery_info.port}"
+        )
+        self._properties = discovery_info.properties
+
+        if not have_mac:
+            self._hub = VegeHub(device_ip)
+
+            await self._hub.retrieve_mac_address()
+
+            if len(self._hub.mac_address) <= 0:
+                _LOGGER.error("Failed to get device config from %s", device_ip)
+                return self.async_abort(reason="cannot_connect")
+
+            try:
+                # Check to see if this MAC address is already in the list.
+                entry = list(ip_dict.keys())[
+                    list(ip_dict.values()).index(self._hub.mac_address)
+                ]
+                if entry:
+                    # If it's already in the list, then it is connected to another
+                    # IP address. Remove that entry.
+                    ip_dict.pop(entry)
+            except ValueError:
+                _LOGGER.info("Zeroconf found new device at %s", device_ip)
+
+            # Add a new entry to the list of IP:MAC pairs that we have seen
+            ip_dict[device_ip] = self._hub.mac_address
+        else:
+            self._hub = VegeHub(device_ip, mac_address=ip_dict[device_ip])
+
+        # Check if this device already exists
+        await self.async_set_unique_id(self._hub.mac_address)
+        self._abort_if_unique_id_configured()
+
+        self.context.update(
+            {
+                "title_placeholders": {"host": self._hostname + " (" + device_ip + ")"},
+                "configuration_url": (self._config_url),
+            }
+        )
```

Comment:
I will go ahead and pull this out to simplify the PR, but something similar will need to go back in eventually. Your addition definitely simplifies things, but it doesn't cover the case where there are VegeHubs on the network that are not intended to be set up in HA.

Any time a VegeHub device, that isn't already set up, joins the local network, the integration will ask it for its MAC address. VegeHubs go to sleep and wake up on a schedule, so any time they wake up and reconnect, the integration will ask them for their MAC address again. Requesting data from a VegeHub keeps it awake for further requests, so battery life will suffer.

Eventually we will need something that keeps track of VegeHubs that have been seen before so the integration doesn't keep requesting their MAC address every time they wake up.

Reply:
I thought so, but my suggestion would be to not ask for the MAC address automatically at all, but only after user interaction (if that makes sense). Otherwise we can try to do this properly, but this felt a bit all over the place tbh.

Reply:
Removed in commit f07414bed09b7842ff5955399c50e733d5cec36c

Reply:
I just talked with my boss about it. We're going to actually change the firmware on the VegeHubs so that there is an API call to quickly grab the MAC address that doesn't wake the device. That should solve the problem with no additional mechanics needed in the integration.

Reply:
This change was made in the device firmware and is tested and working.